diff --git a/config.def.h b/config.def.h
index 1c0b587..ed2c9e1 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */

 /* appearance */
+static const double moveresizeopacity = 0.5;    /* client opacity when being moved or resized */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
diff --git a/dwm.c b/dwm.c
index 4465af1..35cef07 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1140,11 +1140,15 @@ movemouse(const Arg *arg)
        Monitor *m;
        XEvent ev;
        Time lasttime = 0;
+       double prevopacity;

        if (!(c = selmon->sel))
                return;
        if (c->isfullscreen) /* no support moving fullscreen windows by mouse */
                return;
+
+       prevopacity = c->opacity;
+       opacity(c, moveresizeopacity);
        restack(selmon);
        ocx = c->x;
        ocy = c->y;
@@ -1190,6 +1194,7 @@ movemouse(const Arg *arg)
                selmon = m;
                focus(NULL);
        }
+       opacity(c, prevopacity);
 }

 Client *
@@ -1295,11 +1300,15 @@ resizemouse(const Arg *arg)
        Monitor *m;
        XEvent ev;
        Time lasttime = 0;
+       double prevopacity;

        if (!(c = selmon->sel))
                return;
        if (c->isfullscreen) /* no support resizing fullscreen windows by mouse */
                return;
+
+       prevopacity = c->opacity;
+       opacity(c, moveresizeopacity);
        restack(selmon);
        ocx = c->x;
        ocy = c->y;
@@ -1342,6 +1351,7 @@ resizemouse(const Arg *arg)
                selmon = m;
                focus(NULL);
        }
+       opacity(c, prevopacity);
 }

 void
